### PR TITLE
[native] Fix result for empty global grouping sets

### PR DESCRIPTION
**Pull request by aditi-pandit**
## Description
Empty global grouping sets return a row with the default global aggregation result.

This is implemented using special metadata in the AggregationNode. 
Use these new fields recently added to Velox in
 https://github.com/facebookincubator/velox/pull/7112.

## Motivation and Context
Fixes incorrect results for global grouping sets on empty input

 `SELECT orderkey FROM orders WHERE orderkey < 0 GROUP BY CUBE (orderkey)`

returned 0 rows before. Now it returns a single row with a null orderkey

Fixes https://github.com/prestodb/presto/issues/20911

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java [hunk 1]
```diff
@@ -135,6 +135,29 @@ public void testMixedDistinctAggregations()
         assertQuery(session, "SELECT max(orderstatus), COUNT(orderkey), sum(DISTINCT orderkey) FROM orders");
     }
 
+    @Test
+    public void testEmptyGroupingSets()
+    {
+        // This should return a single row with the global aggregation.
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java:141]: nit: "This short return" -> "Returns a single..."

ditto other places

**Review by mbasmanova — COMMENTED** [review]
Looks good, but please update commit message to match the guidelines. Wrap at 72 characters. Remove "This PR updates Prestissimo to use these."

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java (hunk 1 above)
Comment (aditi-pandit) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java]: Done.

**Comment by aditi-pandit** [comment]
@mbasmanova : Have addressed your comments. PTAL.

**Comment by mbasmanova** [comment]
@aditi-pandit Aditi, somehow I don't see commit message updated. I'm looking at https://github.com/prestodb/presto/pull/21288/commits

**Comment by aditi-pandit** [comment]
> @aditi-pandit Aditi, somehow I don't see commit message updated. I'm looking at https://github.com/prestodb/presto/pull/21288/commits

@mbasmanova : I redid this now. PTAL.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java [hunk 2]
```diff
@@ -135,6 +135,29 @@ public void testMixedDistinctAggregations()
...
+        // Returns  a single row with the global aggregation.
+        assertQuery("SELECT count(orderkey) FROM orders WHERE orderkey < 0 GROUP BY GROUPING SETS (())");
+
+        // Returns 2 rows with global aggregation for the global grouping sets.
+        assertQuery("SELECT count(orderkey) FROM orders WHERE orderkey < 0 GROUP BY GROUPING SETS ((), ())");
+
+        // Returns a single row with the global aggregation. There are no rows for the orderkey group.
+        assertQuery("SELECT count(orderkey) FROM orders WHERE orderkey < 0 GROUP BY GROUPING SETS ((orderkey), ())");
+
+        // This is a shorthand for the above query. It returns a single row with the global aggregation.
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java:150]: nit: "A shorthand for the above query. Returns ..."

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java [hunk 3]
```diff
@@ -135,6 +135,29 @@ public void testMixedDistinctAggregations()
...
+        assertQuery("SELECT count(orderkey) FROM orders WHERE orderkey < 0 GROUP BY CUBE (orderkey)");
+
+        assertQuery("SELECT count(orderkey) FROM orders WHERE orderkey < 0 GROUP BY ROLLUP (orderkey)");
+
+        // The following return a single row with NULL orderkey.
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java:155]: nit: "Returns ..."

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Pull request merged**